src/Pages/Expenses.cshtml.cs

```diff
@@ -7,20 +7,12 @@ namespace ePiggyWeb.Pages
 {
     public class ExpensesModel : PageModel
     {
-        public DataEntries Expenses { get; set; }
+        public EntryList Expenses { get; set; }
 
         public void OnGet()
         {
-
-            Expenses = new DataEntries();
-            using var context = new DatabaseContext();
-            var expense = context.Expenses; // define query
-            foreach (var item in expense.Where(x => x.UserId == 0)) // query executed and data obtained from database
-            {
-                var newExpense = new DataEntry(item.Id, item.UserId, item.Amount, item.Title, item.Date, item.IsMonthly,
-                    item.Importance);
-                Expenses.Add(newExpense);
-            }
+            var dataManager = new DataManager();
+            Expenses = dataManager.Expenses.EntryList;
```

Comment:
As I can understand these 2 lines give the same result as all deleted ones? If so it's very neat optimisation. Is it some sort related to that thing called dependency injection?

Reply:
To the first part of the question, yes, the DataManager represents the old Data class, but now I've moved the reading from database code into the constructor. Also I haven't made the pull request, but I also implemented Lazy initialization for the variables in DataManager, so now we will only get the data from database only when we actually call the lists we need.